editor/animation_track_editor.cpp

```diff
@@ -306,10 +306,14 @@ bool AnimationTrackKeyEdit::_set(const StringName &p_name, const Variant &p_valu
 
 				setting = true;
 				undo_redo->create_action(TTR("Animation Change Keyframe Value"), UndoRedo::MERGE_ENDS);
-				int prev = animation->bezier_track_get_key_handle_mode(track, key);
-				undo_redo->add_do_method(this, "_bezier_track_set_key_handle_mode", animation.ptr(), track, key, value);
-				undo_redo->add_undo_method(this, "_bezier_track_set_key_handle_mode", animation.ptr(), track, key, prev);
+				int prev_mode = animation->bezier_track_get_key_handle_mode(track, key);
+				Vector2 prev_in_handle = animation->bezier_track_get_key_in_handle(track, key);
+				Vector2 prev_out_handle = animation->bezier_track_get_key_out_handle(track, key);
+				undo_redo->add_do_method(editor, "_bezier_track_set_key_handle_mode", animation.ptr(), track, key, value);
 				undo_redo->add_do_method(this, "_update_obj", animation);
+				undo_redo->add_undo_method(editor, "_bezier_track_set_key_handle_mode", animation.ptr(), track, key, prev_mode);
+				undo_redo->add_undo_method(animation.ptr(), "bezier_track_set_key_in_handle", track, key, prev_in_handle);
+				undo_redo->add_undo_method(animation.ptr(), "bezier_track_set_key_out_handle", track, key, prev_out_handle);
```

Comment:
`_bezier_track_set_key_handle_mode` seems to have a check for null animation, which means it expects for some reason that the animation can be freed. This code does not have one. Can't say how safe is that.